### PR TITLE
Stop recognizing UTF16_Encoder.encodeUTF16 methods

### PR DESCRIPTION
**Pull request by knn-k**
This commit removes the code for encodeUTF16Big() and encodeUTF16Little() methods in sun.nio.cs.UTF16_Encoder from method recognition, and removes related helper functions.

### CODE REVIEW

**Comment by knn-k** [comment]
`sun.nio.cs.UTF_16$Encoder` no longer has the methods `encodeUTF16Big()` and `encodeUTF16Little()`.
See Issue #20520.

**Review by hzongaro — COMMENTED** [review]
It think the changes look good.  May I ask you to correct the name of the class in the commit message?  I believe it was named `sun.nio.cs.UTF16_Encoder`

File: runtime/compiler/env/j9method.cpp [hunk 1]
```diff
@@ -3779,8 +3777,6 @@ void TR_ResolvedJ9Method::construct()
       {x(TR::sun_nio_cs_ext_SBCS_Decoder_decodeSBCS,              "decodeSBCS",           "([BII[CI[C)I")},
       {x(TR::sun_nio_cs_UTF_8_Encoder_encodeUTF_8,                "encodeUTF_8",     "([CII[BI)I")},
       {x(TR::sun_nio_cs_UTF_8_Decoder_decodeUTF_8,                "decodeUTF_8",          "([BII[CI)I")},
-      {x(TR::sun_nio_cs_UTF_16_Encoder_encodeUTF16Big,            "encodeUTF16Big",       "([CII[BI)I")},
-      {x(TR::sun_nio_cs_UTF_16_Encoder_encodeUTF16Little,         "encodeUTF16Little",    "([CII[BI)I")},
```
[the remark this answers is not in the document]
Comment (knn-k) [reply runtime/compiler/env/j9method.cpp:3785]: I removed the entry from `class24`.

**Comment by knn-k** [comment]
I corrected the class name in the commit title and the commit message.

**Comment by knn-k** [comment]
The test failure on x390x_linux looks the same as Issue #13756.

**Review by hzongaro — APPROVED** [review]
Looks good.  Thanks!

**Comment by knn-k** [comment]
Shall I split the PRs into smaller ones so that they don't require a coordinated merge?

**Comment by hzongaro** [comment]
> Shall I split the PRs into smaller ones so that they don't require a coordinated merge?

No worries!  Let's just go ahead with these PRs as they are.

Windows builds are out of commission at the moment, so those failures are expected.  I will restart the other failing builds.

Jenkins test sanity.functional xmac jdk8 depends eclipse-omr/omr#7548

**Comment by hzongaro** [comment]
Jenkins test sanity.openjdk xmac jdk17,jdk21 depends eclipse-omr/omr#7548

**Comment by hzongaro** [comment]
Jenkins test sanity.functional plinux jdk11,jdk21 depends eclipse-omr/omr#7548

**Comment by hzongaro** [comment]
Failures:

- [JDK 8 sanity.openjdk zLinux](https://openj9-jenkins.osuosl.org/job/Test_openjdk8_j9_sanity.openjdk_s390x_linux_Personal/61/)
    - Failure running `java/util/zip/CloseInflaterDeflaterTest.java` has been encountered intermittently in issue https://github.com/eclipse-openj9/openj9/issues/14948#issuecomment-1206441095
- [JDK 8 sanity.openjdk aix](https://openj9-jenkins.osuosl.org/job/Test_openjdk8_j9_sanity.openjdk_ppc64_aix_Personal/53/)
   - Failure running `java/util/logging/CheckZombieLockTest.java` looks like known issue #9186
- [JDK 8 sanity.functional xLinux](https://openj9-jenkins.osuosl.org/job/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal/267/)
    - Failure running `testJitserverArguments_0` appears to be due to issue #18599
- [JDK 21 sanity.openjdk aix](https://openj9-jenkins.osuosl.org/job/Test_openjdk21_j9_sanity.openjdk_ppc64_aix_Personal/73/)
    - Failure running `jdk_security2_0` appears to be due to issue #19962
    - Failure running `java/lang/Thread/virtual/stress/Skynet.java` appears to be due to known issue https://github.com/eclipse-openj9/openj9/issues/19249#issuecomment-2049641982
- [JDK 21 sanity.functional aarch64 Linux](https://openj9-jenkins.osuosl.org/job/Test_openjdk21_j9_sanity.functional_aarch64_linux_Personal/286/)
    - Failure running `cmdLineTester_criu_nonPortableRestore_10` appears to be known issue #19845

**Comment by hzongaro** [comment]
The only failures appear to be due to known issues.  I will perform the coordinated merge with eclipse-omr/omr#7548 as soon as the same commit appears at the tips of the `openj9` and `master` branches of the `eclipse-openj9/openj9-omr` repository and the `master` branch of `eclipse-omr/omr` repository.

**Comment by knn-k** [comment]
My apologies.
I realized that the `sun.nio.cs.UTF16_Encoder` methods are still used in IBM SDK Java 8.  I thought the class name was `UTF_16` from the enum symbol `sun_nio_cs_UTF_16_Encoder_encodeUTF16*`, and did not check `UTF16_Encoder`.  I should have noticed earlier when @hzongaro added this comment: https://github.com/eclipse-openj9/openj9/pull/20613#pullrequestreview-2461861919
I am preparing PRs for reverting the changes by this PR and https://github.com/eclipse-omr/omr/pull/7548.

**Comment by knn-k** [comment]
I opened PRs #20722, https://github.com/eclipse-omr/omr/pull/7577, and https://github.com/eclipse-omr/omr/pull/7578 for reverting the changes of `UTF16_Encoder.encodeUTF16` methods.

**Pull request merged**